codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -577,42 +578,47 @@ private class ServerHttpProtocolImplGenerator(
         if (pathBindings.isEmpty()) {
             return
         }
-        val pattern = StringBuilder()
         val httpTrait = httpBindingResolver.httpTrait(operationShape)
-        httpTrait.uri.segments.forEach {
-            pattern.append("/")
-            if (it.isLabel) {
-                pattern.append("(?P<${it.content}>")
-                if (it.isGreedyLabel) {
-                    pattern.append(".+")
-                } else {
-                    pattern.append("[^/]+")
```

Comment:
Writing my thoughts down here. I guess done this way because anecdotally servers accept empty path segments e.g. https://github.com//awslabs///////smithy-typescript///commits works fine.

But when extracting labels, I'm not sure about whether this behavior should be the correct one, and I don't see it specified in the Smithy spec. I'm thinking of possible ambiguous scenarios e.g. say the URI pattern is `/foo/{label}/bar` and the server receives a request with URI path `/foo//bar`. Should the `label` field be bound to an empty string? Or should we strip away the empty segment and reject the request?

Reply:
My opinion on this and interpretation of the Smithy spec is that we were indeed doing it wrong and empty path segments **should not** be stripped away, but I've opened an issue with the Smithy team to clarify the spec: https://github.com/awslabs/smithy/issues/1024